### PR TITLE
[YUNIKORN-2396] dependency sync between k8shim and core

### PR DESCRIPTION
**Pull request by brandboat**
### What is this PR for?
upgrade github.com/google/uuid to 1.6.0
upgrade github.com/google/go-cmp to 0.6.0
upgrade github.com/prometheus/client_golang to v1.16.0 and its dependencies in https://github.com/apache/yunikorn-core/pull/786/commits/948eb3d9a23d202aef55b25604c3e0d46a1f4536

### What type of PR is it?
* [x] - Improvement

### Todos
N/A

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2396

### How should this be tested?
N/A

### Screenshots (if appropriate)
N/A

### Questions:
N/A

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -24,14 +24,14 @@ go 1.21
...
-	github.com/google/uuid v1.3.0
+	github.com/google/go-cmp v0.6.0
+	github.com/google/uuid v1.6.0
 	github.com/julienschmidt/httprouter v1.3.0
 	github.com/looplab/fsm v1.0.1
 	github.com/opentracing/opentracing-go v1.2.0
-	github.com/prometheus/client_golang v1.13.1
-	github.com/prometheus/client_model v0.2.0
-	github.com/prometheus/common v0.37.0
+	github.com/prometheus/client_golang v1.16.0
```
Comment (wilfred-s) [inline go.mod:32]: This jumps over a major incompatible version 0.14. Upgrade is logged as  YUNIKORN-2394.

**Review by wilfred-s — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (chia7712) [reply go.mod:32]: oh I did not notice that incompatible change. Big thanks for reminder!

**Comment by chia7712** [comment]
@brandboat Could you please rebase the code?

File: go.mod (hunk 1 above)
Comment (brandboat) [reply go.mod]: already rebased to the latest commit and addressed comments, huge thanks ! 😃

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request closed without merging**